easybuild/easyconfigs/d/dcm2niix/dcm2niix-1.0.20180622-GCCcore-6.4.0.eb

```diff
@@ -0,0 +1,29 @@
+easyblock = 'CMakeMake'
+
+name = 'dcm2niix'
+version = '1.0.20180622'
+
+homepage = 'https://github.com/rordenlab/dcm2niix'
+description = """dcm2niix is a designed to convert neuroimaging data from the DICOM format to the NIfTI format."""
+
+toolchain = {'name': 'GCCcore', 'version': '6.4.0'}
+toolchainopts = {'pic': True}
+
+source_urls = ['https://github.com/rordenlab/dcm2niix/archive']
+sources = ['v%(version)s.tar.gz']
+checksums = ['e9f79509f44aac82c9663381f8f4bfb18a9a3c3eb112d418c92629a871bbb13c']
+
+builddependencies = [
+    ('binutils', '2.28'),
+    ('CMake',    '3.9.5'),
+    ('pkg-config', '0.29.2')
+]
+
+configopts = '-DUSE_OPENJPEG=ON '
```

Comment:
Shouldn't `OpenJPEG` be listed as a dependency then?

Reply:
no need. It seems in this newest version they already include the OpenJPEG inside. So you only need to switch on the option.

Reply:
@murfreesboro Do you know if it's just a vanilla OpenJPEG they ship? Do they provide an option to use an already installed OpenJPEG? If so, that's highly preferable...

Reply:
OK, the downloading part is definitely not OK, since that prevents offline installation, we should prevent that if we can.

Reply:
according to the readme.md the author states that this version of dcm2niix sticks to the OpenJPEG 2.1.0, with the higher versions of OpenJPEG it requires "custom installation" (You can build dcm2niix with JPEG2000 decompression support using OpenJPEG 2.1.0. If you want to use a more recent version of OpenJPEG, it must be custom-compiled with `DOPJ_DISABLE_TPSOT_FIX` compiler flag.)

Reply:
I would then suggest to go ahead and link it against `OpenJPEG-2.1.0`

Reply:
This is the place that "DOPJ_DISABLE_TPSOT_FIX" used in the cmake file of OpenJPEG:

option(OPJ_DISABLE_TPSOT_FIX "Disable TPsot==TNsot fix. See https://github.com/uclouvain/openjpeg/issues/254." OFF)
if(OPJ_DISABLE_TPSOT_FIX)
  add_definitions(-DOPJ_DISABLE_TPSOT_FIX)
endif()

It looks like this is a bug fix since the version of 2.0.0. So the information given by the author is unclear that why since the version > 2.1.0 this option need to be enforced. I agree that we probably use OpenJPEG 2.3.0 with this option.

Reply:
I guess it makes sense to go forward with OpenJPEG 2.1.0, since if we use 2.3.0 we need a custom compilation of it anyway.

However, that does mean our test that checks that we only use a single "variant" of a software package (`OpenJPEG` in this case) as dependency per toolchain generation will fail.

We can circumvent that if we only include `OpenJPEG` 2.1.0 as a build-only dependency, and statically link it, since in that case there is no potential version conflict with other modules that also load `OpenJPEG` as a dependency.

Is that an option @murfreesboro?

Reply:
this is a good idea @boegel. I will fix the eb file and static link to the OpenJPEG 2.1.0.

Reply:
I tried your idea @boegel, however finally it turns out that the program does not support the static linking for the  OpenJPEG 2.1.0. This library always goes in as dynamic library. For cmake options although the dcm2niix support the USE_STATIC_RUNTIME, this option only works for the system library like libstdc++ and libgcc. So finally I add the OpenJPEG 2.1.0 back to the dependency.

Reply:
@boegel it seems the new eb file verification failed. However I went through the failing message and I can not get the point. It seems the style check passed. Is there something wrong with the current eb file?

Reply:
@murfreesboro It looks like one of the tests failed due to a fluke in Travis CI. Usually my bot takes care of restarting broken builds, but it was temporarily down last night, it seems like you "got lucky".

I've restarted the failing test, should be fine.

Thanks a lot for following up on this!